sqlmesh/core/dialect.py

```diff
@@ -755,12 +755,21 @@ def select_from_values_for_batch_range(
         exp.alias_(exp.cast(column, to=kind), column, copy=False)
         for column, kind in columns_to_types.items()
     ]
+
+    if not values:
+        where = exp.false()  # Force a zero-row output
+        values = [(exp.null(),) * len(columns_to_types)]  # Don't generate an empty VALUES clause
+    else:
+        where = None
+        values = values[batch_start:batch_end]
+
     values_exp = exp.values(
-        [tuple(transform_values(v, columns_to_types)) for v in values[batch_start:batch_end]],
+        [tuple(transform_values(v, columns_to_types)) for v in values],
         alias=alias,
         columns=columns_to_types,
     )
-    return exp.select(*casted_columns).from_(values_exp, copy=False)
+
+    return exp.select(*casted_columns).from_(values_exp).where(where)
```

Comment:
since this is taking a where statement here with values, this could be large, so we want to copy=False

Reply:
yeah fair point, I added the `copy=False` kwarg back and did another refactor